vhost-user-backend/src/backend.rs

```diff
@@ -89,9 +89,14 @@ pub trait VhostUserBackend: Send + Sync {
     /// Set handler for communicating with the frontend by the gpu specific backend communication
     /// channel.
     ///
-    /// This method only exits when the crate feature gpu-socket is enabled, because this is only
-    /// useful for a gpu device.
-    fn set_gpu_socket(&self, _gpu_backend: GpuBackend);
+    /// This function return a `Result`, returning an error if the backend does not implement this
+    /// function.
+    fn set_gpu_socket(&self, _gpu_backend: GpuBackend) -> Result<()> {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Unsupported,
+            "back end requires set_gpu_socket to be implemented",
```

Comment:
I'd follow `check_device_state()`, so something like `back end does not support gpu socket`.